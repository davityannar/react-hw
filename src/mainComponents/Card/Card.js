import React, {Component} from 'react';
import './Card.scss';
import { connect } from 'react-redux';
// import {miCardCucadrel} from '../../actions/actionCreators'
import { withRouter } from 'react-router-dom'
import queryString  from 'query-string'
import CardItem from '../CardItem/CardItem';
import NoFilmsFound from "../NoFilmsFound/NoFilmsFound";

import {asyncHandlingSubmitFormAction} from '../../actions/actionCreators'

export class Card extends Component {
    constructor(props) {
      super(props)
    }

    componentDidMount() {
      const values = queryString.parse(this.props.location.search)
      if (values.limit) {
        const {search, searchBy, sortBy} = values
        this.props.showApiData(search, searchBy, sortBy);
      }
    }
    
    render() {
      
        let pageContent = this.props.apiData.map( content => <CardItem key={content.id} 
            movieId={content.id}
            onCustomClick={this.props.addingContentToOneCardComponent} 
            imgSrc={content.poster_path} 
            movieVote={content.vote_average} 
            movieName={content.title} 
            movieAge={content.release_date.slice(0, 4)} 
            movieGenre={content.genres.join(', ')}></CardItem>)
      return (
        <div className='card-wrapper'>
           {pageContent}
        </div>
      )
      
    }
  }

  const mapStateToProps = (state) => ({apiData: state.search.apiData})
  const mapDispatchToProps = (dispatch) => ({
    showApiData(searchText, searchBy, sortBy) {
      dispatch(asyncHandlingSubmitFormAction(searchText, searchBy, sortBy));
    }
  })

  
  export default withRouter(connect(mapStateToProps, mapDispatchToProps)(Card))







