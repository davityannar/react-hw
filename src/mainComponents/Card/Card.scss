.card-wrapper {
    display: flex;
    justify-content: flex-start;
    flex-wrap: wrap;
    flex-grow: 1;
    width: 1200px;
    margin: 0 auto;
}

.no-films-found-wrapper {
    flex-grow: 1;
  }
  
  .no-films-found-wrapper {
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 48px;
    font-weight: bold;
  }