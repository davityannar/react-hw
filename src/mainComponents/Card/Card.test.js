import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Tests Cards container', () => {
  it('Matching snapshot', () => {

    const props = {
        data: 
            {
                budget: 0,
                genres: ["Mystery", "Thriller"],
                id: 412302,
                overview: "and confront.",
                poster_path: "https://image.tmdb.org/t/p/w500/oIltQs7MPk7VQFG3DJfgC63mShU.jpg",
                release_date: "2018-03-30",
                revenue: 0,
                runtime: 92,
                tagline: "",
                title: "Gemini",
                vote_average: 10,
                vote_count: 3,
            }
        
    };

    const container = shallow(<Card { ...props }/>)

    expect(container).toMatchSnapshot();
  })
})