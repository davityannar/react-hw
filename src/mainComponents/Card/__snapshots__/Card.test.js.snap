// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Cards container Matching snapshot 1`] = `
<div>
  <div
    className="no-films-found-wrapper"
  >
    <p
      className="no-films-found-text"
    >
      No Films Found
    </p>
  </div>
  <div
    className="card-wrapper"
  />
</div>
`;
